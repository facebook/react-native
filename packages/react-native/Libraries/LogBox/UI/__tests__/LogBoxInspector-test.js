/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../../jest/renderer');
const LogBoxLog = require('../../Data/LogBoxLog').default;
const LogBoxInspector = require('../LogBoxInspector').default;
const React = require('react');

// Mock child components because we are interested in snapshotting the behavior
// of `LogBoxInspector`, not its children.
jest.mock('../LogBoxInspectorBody', () => ({
  __esModule: true,
  default: 'LogBoxInspectorBody',
}));
jest.mock('../LogBoxInspectorFooter', () => ({
  __esModule: true,
  default: 'LogBoxInspectorFooter',
}));
jest.mock('../LogBoxInspectorHeader', () => ({
  __esModule: true,
  default: 'LogBoxInspectorHeader',
}));
jest.mock('../../Data/LogBoxData', () => ({
  __esModule: true,
  symbolicateLogLazy: () => {},
  symbolicateLogNow: () => {},
}));

const logs = [
  new LogBoxLog({
    level: 'warn',
    isComponentError: false,
    message: {
      content: 'Some kind of message (first)',
      substitutions: [],
    },
    stack: [],
    category: 'Some kind of message (first)',
    componentStack: [],
  }),
  new LogBoxLog({
    level: 'error',
    isComponentError: false,
    message: {
      content: 'Some kind of message (second)',
      substitutions: [],
    },
    stack: [],
    category: 'Some kind of message (second)',
    componentStack: [],
  }),
  new LogBoxLog({
    level: 'fatal',
    isComponentError: false,
    message: {
      content: 'Some kind of message (third)',
      substitutions: [],
    },
    stack: [],
    category: 'Some kind of message (third)',
    componentStack: [],
  }),
];

describe('LogBoxContainer', () => {
  it('should render null with no logs', async () => {
    const output = await render.create(
      <LogBoxInspector
        onDismiss={() => {}}
        onMinimize={() => {}}
        onChangeSelectedIndex={() => {}}
        logs={[]}
        selectedIndex={0}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render warning with selectedIndex 0', async () => {
    const output = await render.create(
      <LogBoxInspector
        onDismiss={() => {}}
        onMinimize={() => {}}
        onChangeSelectedIndex={() => {}}
        logs={logs}
        selectedIndex={0}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render fatal with selectedIndex 2', async () => {
    const output = await render.create(
      <LogBoxInspector
        onDismiss={() => {}}
        onMinimize={() => {}}
        onChangeSelectedIndex={() => {}}
        logs={logs}
        selectedIndex={2}
        fatalType="fatal"
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
