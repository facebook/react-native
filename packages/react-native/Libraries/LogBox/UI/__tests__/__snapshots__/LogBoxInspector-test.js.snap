// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxContainer should render fatal with selectedIndex 2 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "flex": 1,
    }
  }
>
  <LogBoxInspectorHeader
    level="fatal"
    onSelectIndex={[Function]}
    selectedIndex={2}
    total={3}
  />
  <LogBoxInspectorBody
    log={
      LogBoxLog {
        "category": "Some kind of message (third)",
        "codeFrame": undefined,
        "componentStack": Array [],
        "componentStackType": "legacy",
        "count": 1,
        "extraData": undefined,
        "isComponentError": false,
        "level": "fatal",
        "message": Object {
          "content": "Some kind of message (third)",
          "substitutions": Array [],
        },
        "onNotificationPress": undefined,
        "stack": Array [],
        "symbolicated": Object {
          "error": null,
          "stack": null,
          "status": "NONE",
        },
        "symbolicatedComponentStack": Object {
          "componentStack": null,
          "error": null,
          "status": "NONE",
        },
        "type": undefined,
      }
    }
    onRetry={[Function]}
  />
  <LogBoxInspectorFooter
    level="fatal"
    onDismiss={[Function]}
    onMinimize={[Function]}
  />
</View>
`;

exports[`LogBoxContainer should render null with no logs 1`] = `null`;

exports[`LogBoxContainer should render warning with selectedIndex 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "flex": 1,
    }
  }
>
  <LogBoxInspectorHeader
    level="warn"
    onSelectIndex={[Function]}
    selectedIndex={0}
    total={3}
  />
  <LogBoxInspectorBody
    log={
      LogBoxLog {
        "category": "Some kind of message (first)",
        "codeFrame": undefined,
        "componentStack": Array [],
        "componentStackType": "legacy",
        "count": 1,
        "extraData": undefined,
        "isComponentError": false,
        "level": "warn",
        "message": Object {
          "content": "Some kind of message (first)",
          "substitutions": Array [],
        },
        "onNotificationPress": undefined,
        "stack": Array [],
        "symbolicated": Object {
          "error": null,
          "stack": null,
          "status": "NONE",
        },
        "symbolicatedComponentStack": Object {
          "componentStack": null,
          "error": null,
          "status": "NONE",
        },
        "type": undefined,
      }
    }
    onRetry={[Function]}
  />
  <LogBoxInspectorFooter
    level="warn"
    onDismiss={[Function]}
    onMinimize={[Function]}
  />
</View>
`;
